google/cloud/spanner_v1/_helpers.py

```diff
@@ -460,6 +464,34 @@ def _metadata_with_prefix(prefix, **kw):
     return [("google-cloud-resource-prefix", prefix)]
 
 
+def _retry_on_aborted_exception(
+    func,
+    deadline,
+    allowed_exceptions=None,
+):
+    """
+    Handles retry logic for Aborted exceptions, considering the deadline.
+    Retries the function in case of Aborted exceptions and other allowed exceptions.
+    """
+    attempts = 0
+    while True:
+        try:
+            attempts += 1
+            return func()
+        except Aborted as exc:
+            _delay_until_retry(exc, deadline=deadline, attempts=attempts)
+            continue
+        except Exception as exc:
+            try:
+                retry_result = _retry(func=func, allowed_exceptions=allowed_exceptions)
+                if retry_result is not None:
+                    return retry_result
+                else:
+                    raise exc
+            except Aborted:
+                continue
```

Comment:
I think that we should remove this part entirely. I know that the previous implementation of `Batch` retried this specific `RST_STREAM` error, but that was just a copy-paste from other methods. That error is not relevant for this type of operation.